fetch/fetch.go

```diff
@@ -511,47 +578,36 @@ func (f *Fetch) sendBatch(requests []requestMessage) {
 	if err != nil {
 		f.handleHashError(batch.ID, err)
 	}
-	// try sending batch to some random peer
+
+	// try sending batch to provided peer
 	retries := 0
 	for {
 		if f.stopped() {
-			return
-		}
-
-		if f.net.PeerCount() == 0 {
-			f.log.With().Error("no peers found, unable to send request batch",
-				batch.ID,
-				log.Int("items", len(batch.Requests)))
-			return
+			return nil
 		}
 
-		// get random peer
-		p := GetRandomPeer(f.net.GetPeers())
 		f.log.With().Debug("sending request batch to peer",
 			log.String("batch_hash", batch.ID.ShortString()),
 			log.Int("num_requests", len(batch.Requests)),
 			log.String("peer", p.String()))
-		batch.peer = p
-		f.activeBatchM.Lock()
-		f.activeBatches[batch.ID] = batch
-		f.activeBatchM.Unlock()
-		err := f.net.Request(context.TODO(), p, bytes, f.receiveResponse, errorFunc)
 
-		// if call succeeded, continue to other requests
-		if err != nil {
-			retries++
-			if retries > f.cfg.MaxRetriesForPeer {
-				f.handleHashError(batch.ID, ErrCouldNotSend(fmt.Errorf("could not send message: %w", err)))
-				break
-			}
-			// todo: mark number of fails per peer to make it low priority
-			f.log.With().Warning("could not send message to peer",
-				log.String("peer", p.String()),
-				log.Int("retries", retries))
-		} else {
+		err = f.net.Request(context.TODO(), p, bytes, f.receiveResponse, errorFunc)
+		if err == nil {
 			break
 		}
+
+		retries++
+		if retries > f.cfg.MaxRetriesForPeer {
```

Comment:
in the old code, it's called MaxRetriesForPeer, but it was really trying different peer for each retry.

i wanted to drop this param too, but i guess we can keep it in case of transient network error for the same peer.